### PR TITLE
Permissions setting for the rest of the actions

### PR DESCRIPTION
**Pull request by ytsarev**
Follow up to #990

For some reason https://app.stepsecurity.io/ didn't manage to handle these actions automatically.

Signed-off-by: Yury Tsarev <yury@upbound.io>

### CODE REVIEW

File: .github/workflows/gh-pages.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@ on:
     paths:
       - '**.md'
       - 'docs/**'
+permissions:
+  contents: read
```
Comment (jkremser) [inline .github/workflows/gh-pages.yaml:11]: i think we need write here, the implicit GITHUB_TOKEN is being used by EndBug/add-and-commit

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -6,11 +6,13 @@ on:
       - "v*"
 
 permissions:
-  contents: write
+  contents: read
```
Comment (jkremser) [inline .github/workflows/release.yaml:9]: this will break the next release, the contents: write is required by two other jobs in this workflow

**Review by jkremser — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-pages.yaml (hunk 1 above)
Comment (ytsarev) [reply .github/workflows/gh-pages.yaml:11]: we are setting 
```
    permissions:
      contents: write
```
on a job level

**Review by jkremser — APPROVED** [review]
thanks

**Pull request merged**